server/mdm/microsoft/microsoft_mdm.go

```diff
@@ -275,3 +279,14 @@ func ResolveWindowsMDMAuth(serverURL string) (string, error) {
 func ResolveWindowsMDMManagement(serverURL string) (string, error) {
 	return commonmdm.ResolveURL(serverURL, MDE2ManagementPath, false)
 }
+
+// Encrypt uses pkcs7 to encrypt a raw value using the provided certificate.
+// The returned encrypted value is base64-encoded.
+func Encrypt(rawValue string, cert *x509.Certificate) (string, error) {
```

Comment:
@marcosd4h Do we need the key to be encrypted a specific way in Fleet? The spec mentioned "Encrypt the key using the wstep cert and Base64 encode the key", but nothing more specific. Unless we need it a specific way for some reason, I went with the same way as the macos keys are encrypted (pkcs7 with the cert, in this case the WSTEP cert).

It's nice because if/when we need to send it decrypted, the logic can be the same for both platforms, with just the certificate to use to decrypt that will depend on the platform.